packages/headless/src/controllers/core/tab/headless-core-tab.test.ts

```diff
@@ -0,0 +1,111 @@
+import {configuration, tabSet} from '../../../app/reducers';
+import {getConfigurationInitialState} from '../../../features/configuration/configuration-state';
+import {
+  registerTab,
+  updateActiveTab,
+} from '../../../features/tab-set/tab-set-actions';
+import {buildMockSearchAppEngine, MockSearchEngine} from '../../../test';
+import {buildMockTabSlice} from '../../../test/mock-tab-state';
+import {buildCoreTab, Tab, TabProps} from './headless-core-tab';
+
+describe('Core Tab', () => {
+  let engine: MockSearchEngine;
+  let props: TabProps;
+  let tab: Tab;
+
+  function initTab() {
+    tab = buildCoreTab(engine, props);
+  }
+
+  beforeEach(() => {
+    engine = buildMockSearchAppEngine();
+    props = {
+      options: {
+        expression: '@objecttype==Message',
+        id: 'All',
+      },
+      initialState: {
+        isActive: false,
+      },
+    };
+
+    initTab();
+  });
+
+  it('it adds the correct reducers to engine', () => {
+    expect(engine.addReducers).toHaveBeenCalledWith({
+      configuration,
+      tabSet,
+    });
+  });
+
+  it('when the #id option is an empty string, it throws', () => {
+    props.options.id = '';
+    expect(() => initTab()).toThrow();
+  });
+
+  it('when the #id option is set to default originLevel2 value, it throws', () => {
+    props.options.id = getConfigurationInitialState().analytics.originLevel2;
+    expect(() => initTab()).toThrow();
+  });
+
+  describe('initalization', () => {
+    it('dispatches #registerTab', () => {
+      initTab();
+
+      const {id, expression} = props.options;
+      const action = registerTab({id, expression});
+      expect(engine.actions).toContainEqual(action);
+    });
+
+    it('when isActive is true, it dispatches #updateActiveTab', () => {
+      const {id} = props.options;
+      props.initialState!.isActive = true;
+      initTab();
+
+      expect(engine.actions).toContainEqual(updateActiveTab(id));
+    });
+
+    it('does not throw if initialState is undefined', () => {
```

Comment:
should we test that it initializes correctly and not just that it doesn't throw?